### PR TITLE
Fix Check For num_entries for CpuId #119

### PR DESCRIPTION
**Pull request by gabhijit**
Added a check to verify that the number of entires passed is less than
or equal to `KVM_MAX_CPUID_ENTRIES` (`Kvm.get_cpuid`,=`VcpuFd.get_cpuid2`).

Added test cases for failure scenarios for `get_emulated_cpuid` and
`get_supported_cpuid`.

Renamed param `max_entries_count` to `num_entries` consistent with
`VcpuFd.get_cpuid2`

### CODE REVIEW

**Comment by gabhijit** [comment]
Odd that I do not get the same warnings if I do a local `cargo build` and or `cargo test`? Not sure why those warnings. 

Does commit format require a 'signed-off-by'? line May be that's causing the failure?

File: src/ioctls/system.rs
Comment (acatangiu) [inline src/ioctls/system.rs]: nit: Actually newer kernels return `E2BIG`, you could do the same.

File: src/ioctls/system.rs
Comment (acatangiu) [inline src/ioctls/system.rs]: nit: this is small enough to be another check in the `test_get_supported_cpuid()` test instead of a standalone test

File: src/ioctls/system.rs
Comment (acatangiu) [inline src/ioctls/system.rs]: nit: this is small enough to be another check in the `test_get_emulated_cpuid()` test instead of a standalone test

File: src/ioctls/vcpu.rs
Comment (acatangiu) [inline src/ioctls/vcpu.rs]: nit: same here, `E2BIG` is better

**Review by acatangiu — COMMENTED** [review]
The kernel already does all the validations included in this PR. The overhead here is minimal but also not necessary. I'll leave the final decision to the other maintainers.

File: src/ioctls/system.rs
Comment (gabhijit) [reply src/ioctls/system.rs]: Actually `E2BIG` is returned if the passed number of entries is too low and `ENOMEM` is returned if the passed number of entries is too high. So I believe in this case `ENOMEM` is right - as per [this documentation](https://www.kernel.org/doc/html/latest/virt/kvm/api.html?highlight=kvm%20ioctl)

File: src/ioctls/system.rs
Comment (gabhijit) [reply src/ioctls/system.rs]: Okay will do it there.

File: src/ioctls/system.rs
Comment (gabhijit) [reply src/ioctls/system.rs]: Okay will do it there.

File: src/ioctls/vcpu.rs
Comment (gabhijit) [reply src/ioctls/vcpu.rs]: Same as above. I think `ENOMEM` is correct.

File: src/ioctls/system.rs
Comment (andreeaflorescu) [inline src/ioctls/system.rs:278]: I think we should also update the documentation of the function to specify that it returns an error when `num_entries` > KVM_MAX_CPUID_ENTRIES. Same for `get_supported_cpuid`.

**Review by andreeaflorescu — DISMISSED** [review]

**Review by jiangliu — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**
